core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthChecker.java

```diff
@@ -16,14 +16,50 @@
 
 package com.linecorp.armeria.server.healthcheck;
 
+import java.time.Duration;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.common.CommonPools;
 import com.linecorp.armeria.server.Server;
 
+import io.netty.channel.EventLoop;
+
 /**
  * Determines whether the {@link Server} is healthy. All registered {@link HealthChecker}s must return
  * {@code true} for the {@link Server} to be considered healthy.
  */
 @FunctionalInterface
 public interface HealthChecker {
+
+    /**
+     * Create a {@link HealthChecker} which executing supplied health checker {@link Supplier} on a random
+     * {@link EventLoop} from {@link CommonPools#workerGroup()} after constructing and subsequently with the
+     * given interval.
+     *
+     * @param healthChecker the {@link Supplier} of {@link CompletionStage} that provides the result of health
+     * @param interval the interval between successive executions
+     * @param jitter the rate that used to calculate the lower and upper bound of the backoff delay
+     */
+    static HealthChecker ofFixedRate(Supplier<CompletionStage<Boolean>> healthChecker, Duration interval,
+                                     double jitter) {
+        return new FixedRateHealthChecker(healthChecker, interval, jitter, CommonPools.workerGroup().next());
+    }
+
+    /**
+     * Create a {@link HealthChecker} which executing supplied health checker {@link Supplier} on a random
+     * {@link EventLoop} from {@link CommonPools#workerGroup()} after constructing and subsequently with the
+     * given delay between the completion of supplied {@link CompletionStage} and the commencement of the next.
+     *
+     * @param healthChecker the {@link Supplier} of {@link CompletionStage} that provides the result of health
+     * @param delay  fixed delay between attempts
+     * @param jitter the rate that used to calculate the lower and upper bound of the backoff delay
+     */
+    static HealthChecker ofFixedDelay(Supplier<CompletionStage<Boolean>> healthChecker, Duration delay,
+                                      double jitter) {
```

Comment:
How about limiting the scope of the `jitter`?
`checkArgument(0.0 <= jitter && jitter <= 1.0, ...)`

Reply:
How about adding a variation that takes an `EventExecutor`?